flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/delegation/DefaultExecutorTest.java

```diff
@@ -105,6 +72,35 @@ public void testJobName() {
                 "Custom Name");
     }
 
+    @Test
+    public void testDefaultBatchProperties() {
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        final Executor executor = new DefaultExecutor(env);
+
+        final List<Transformation<?>> dummyTransformations =
+                Collections.singletonList(
+                        env.fromElements(1, 2, 3)
+                                .addSink(new DiscardingSink<>())
+                                .getTransformation());
+
+        final Configuration configuration = new Configuration();
+        configuration.set(ExecutionOptions.RUNTIME_MODE, RuntimeExecutionMode.BATCH);
+
+        final StreamGraph streamGraph =
+                (StreamGraph)
+                        executor.createPipeline(
+                                dummyTransformations, configuration, "Default Name");
+
+        assertTrue(streamGraph.getExecutionConfig().isObjectReuseEnabled());
+        assertEquals(0, streamGraph.getExecutionConfig().getLatencyTrackingInterval());
+        assertTrue(streamGraph.isChainingEnabled());
+        assertFalse(streamGraph.isAllVerticesInSameSlotSharingGroupByDefault());
+        assertFalse(streamGraph.getCheckpointConfig().isCheckpointingEnabled());
```

Comment:
Should we assert watermark interval and buffer timeout also have been disabled too?

Reply:
Watermark interval has a value but this value should not be read, because watermarks are disabled for `SourceFunction`. This is tested in `org.apache.flink.table.planner.plan.nodes.exec.TransformationsTest#testLegacyBatchSource`. Buffer timeout is disabled in https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java#L361